src/site/content/en/blog/websocketstream/index.md

```diff
@@ -0,0 +1,332 @@
+allows JavaScript to programmatically access streams of data chunks received over the network
+and process them as desired by the developer.
+An important concept in the context of streams is
+[backpressure](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API/Concepts#Backpressure).
+This is the process by which a single stream or a pipe chain
+regulates the speed of reading or writing.
+When the stream itself or a stream later in the pipe chain is still busy
+and isn't yet ready to accept more chunks,
+it sends a signal backwards through the chain to slow down delivery as appropriate.
+
+### The Problem with the current WebSocket API
+
+#### Applying backpressure to received messages is impossible
+
+With the current WebSocket API, reacting to a message happens via the
+[`WebSocket.onmessage`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/onmessage)
+property, an `EventHandler` that is called when a message is received from the server.
+
+Let's assume you had an application that needed to perform some heavy data crunching operations
+whenever a new message is received.
+You would probably set up the flow similar to the code in the example below,
+and since you `await` the result of the `process()` call, you should be good, right?
+
+```js
+// A heavy data crunching operation.
+const process = async (data) => {
+  return new Promise((resolve) => {
+    window.setTimeout(() => {
+      console.log('WebSocket message processed:', data);
+      return resolve('done');
+    }, 1000);
+  });
+};
+
+webSocket.onmessage = async (event) => {
+  const data = event.data;
+  // Await the result of the processing step in the message handler.
+  await process(data);
+};
+```
+
+Wrong! The problem with the current WebSocket API is that there is no way to apply backpressure.
+When messages arrive faster than the `process()` method can handle them,
+the render process will either fill up memory by buffering those messages,
+become unresponsive due to 100% CPU usage, or both.
+
+#### Applying backpressure to sent messages is non-ergonomic
+
+Applying backpressure to sent messages is possible, but involves polling the
+[`WebSocket.bufferedAmount`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/bufferedAmount)
+property, which is inefficient and non-ergonomic.
+This read-only property returns the number of bytes of data that have been queued
+using calls to
+[`WebSocket.send()`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/send),
+but not yet transmitted to the network.
+This value resets to zero once all queued data has been sent,
+but if you keep calling `WebSocket.send()`,
+it will continue to climb.
+
+## What is the WebSocketStream API? {: #what }
+
+The new WebSocketStream API deals with the problem of non-existent or non-ergonomic backpressure
+applicability by integrating Streams with the WebSocket API.
+This means backpressure can be applied "for free", without any extra hoops.
```

Comment:
```suggestion
This means backpressure can be applied "for free", without any extra cost.
```

Reply:
I don't usually quibble with a writers choice of idiom. Shortening 'jumping through hoops' to 'hoops' may cause problems for readers who are not fluent in English. Cost is easier to translate than a figure of speech is.